src/daemon/emitter.cc

```diff
@@ -527,5 +582,63 @@ void Emitter::DoRemoteExecute(const base::WorkerPool& pool,
   }
 }
 
+void Emitter::DoCoordinator(const base::WorkerPool& pool,
+                            std::vector<ResolveFn> resolvers) {
+  net::EndPointPtr end_point;
+  net::ConnectionPtr connection;
+  ui32 sleep_period = 1;
+  auto Sleep = [&sleep_period]() mutable {
+    LOG(INFO) << "Sleeping for " << sleep_period
+              << " seconds before next attempt";
+    std::this_thread::sleep_for(std::chrono::seconds(sleep_period));
+    if (sleep_period < static_cast<ui32>(-1) / 2) {
+      sleep_period <<= 1;
+    }
+  };
+
+  compilation_workers_.reset(new base::WorkerPool(true));
+  while (!pool.IsShuttingDown()) {
+    if (!connection) {
+      for (const ResolveFn& resolver : resolvers) {
```

Comment:
I don't like the concept of round-robin the coordinators in a single thread - at least you don't have a separate sleep time for each one. Let's think a better way. May be setup a worker per coordinator - with a common atomic variable about last update. Run them with a time shift about 5 minutes. If the first one doesn't update config before the second one wakes up, then the second one tries to update config, and so on.